### PR TITLE
Feature/bitcoin crypto

### PR DESCRIPTION
**Pull request by joscas**

### CODE REVIEW

File: domain/src/main/scala/BitcoinCryptography.scala [hunk 1]
```diff
@@ -1,13 +1,13 @@
...
-    "btc-public-key"
-  }
+import org.apache.commons.codec.binary.Hex
+import org.bitcoinj.core.{ECKey, NetworkParameters}
 
-  // TODO: implement me
-  def bitcoinP2SHAddressFromPublicKey(key: String): String = {
-    "btc-public-key"
+trait BitcoinCryptography {
+  def bitcoinAddressFromPublicKey(key: String): String = {
```
Comment (pauloglesby) [inline domain/src/main/scala/BitcoinCryptography.scala:7]: So now we only have one type of address from the bitcoin PK?

Comment (joscas) [reply domain/src/main/scala/BitcoinCryptography.scala:7]: As far as I've been able to see. the `bitcoinj` library takes care of this complexity for us. Not 100% sure yet though but I would take it as a working assumption

Comment (pauloglesby) [reply domain/src/main/scala/BitcoinCryptography.scala:7]: Nice. We'd be able to add `address` as a method on the base sealed trait `PublicKey` in that case 😉

Comment (joscas) [reply domain/src/main/scala/BitcoinCryptography.scala:7]: Indeed

**Pull request merged**